README.md

```diff
@@ -5,21 +5,97 @@ Right now this app only works on android. iOS support is planned in the future.
 
 ## Project setup
 
-_This project assumes that the machine running ths code has JDK, Android SDK and other basic dependencies install to run an Android project._
+The instructions provided below are for macOS. Detailed setup instructions for react-native projects along with dependencies for other operating systems can be found [here](https://reactnative.dev/docs/environment-setup?guide=native&platform=android)
+
+### Node and JDK setup
 
 This app is a react-native based app with java bridges to interface between android core API's and the react-native component.
-Basic setup instructions for react-native projects along with dependencies can be found [here](https://reactnative.dev/docs/environment-setup?guide=native&platform=android)
-Please follow this guide exactly.
+Install node and watchman. If you have already installed Node on your system, make sure it is Node 18 or newer. Watchman is a tool by Facebook for watching changes in the filesystem. It is highly recommended you install it for better performance.
+
+```sh
+brew install node
+brew install watchman
+```
+We recommend installing the OpenJDK distribution called Azul Zulu.
+
+```sh
+brew tap homebrew/cask-versions
+brew install --cask zulu17
+
+# Get path to where cask was installed to double-click installer
+brew info --cask zulu17
+```
+
+After you install the JDK, update your JAVA_HOME environment variable. If you used above steps, JDK will likely be at `/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home`
+
+If you have already installed JDK on your system, we recommend JDK 17. You may encounter problems using higher JDK versions. To make sure you are using the JDK version 17 update the `JAVA_HOME` and `PATH` variable to reflect the same.
```

Comment:
We recommend JDK 17 as higher JDK versions might cause some issues. To make sure that the JDK version 17 is being used, update the `JAVA_HOME` and `PATH` variable to reflect the same.